osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+import sys
+from datetime import date
+
+# Set up django configuration
+import django
+from kobo.client.constants import FINISHED_STATES
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+django.setup()
+
+from kobo.hub.models import Task  # noqa: E402
+
+from osh.hub.scan.models import RetentionPolicySetting  # noqa: E402
+from osh.hub.scan.models import TaskResultsRemoval  # noqa: E402
+
+logger = logging.getLogger("osh.hub.scripts.osh-retention")
+
+
+DEFAULT_RETENTION_DAYS = 30
+
+
+def delete_tasks_results(tasks_to_delete):
+    for task, retention_policy_setting in tasks_to_delete:
+        try:
+            path_to_delete = Task.get_task_dir(task.id)
+            # if folder exists, we delete it. if not, we mark it as deleted as the folder already doesn't exist
+            if os.path.isdir(path_to_delete):
+                shutil.rmtree(path_to_delete)
+                logger.debug('The following folder was deleted: %s', path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+            continue
+
+        # If the tasks results are deleted, we update the database field, so we don't try to delete them again.
+        TaskResultsRemoval.objects.create(task=task, reason=retention_policy_setting)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner and "-" not in owner
+
+
+def main():
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    # Retrieval of all tasks ids that already have the retention policy applied
+    processed_tasks = TaskResultsRemoval.objects.values_list('task', flat=True)
+    # Retrieval of tasks that has not been applied to retention policy to be applied
+    tasks = Task.objects.filter(state__in=FINISHED_STATES).exclude(id__in=processed_tasks)
+
+    # ---------- Section to take parameters from DB
+    try:
+        retention_personal_tasks = RetentionPolicySetting.objects.get(category="PersonalScan")
+    except django.core.exceptions as e:
+        logger.error("Error obtaining PersonalScan value %s.", e)
+        sys.exit(1)
+
+    try:
+        retention_failed_status = RetentionPolicySetting.objects.get(category="FailedStatus")
+    except django.core.exceptions as e:
+        logger.error("Error obtaining FailedStatus value: %s.", e)
```

Comment:
The wrong exception usage mentioned in my last review comment is still not resolved.

Reply:
I changed the exceptions on the new commit

Reply:
Well, you did change the code but it is still broken and it still crashes on my machine... 

```python
...
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/src/osh/hub/scripts/osh-retention", line 110, in <module>
    main()
  File "/src/osh/hub/scripts/osh-retention", line 69, in main
    except django.db.DatababaseError as e:
AttributeError: module 'django.db' has no attribute 'DatababaseError'
```

@jperezdealgaba I don't want to be rude but do you **actually test** any of your changes locally?  You have to understand that OpenScanHub maintenance is not in my job description and I'd love to spend the little time I have for it by doing something actually meaningful and not by repeated testing of your broken code which should have been tested already by yourself before you pushed it to GitHub.

Reply:
I have been triggering the exception in different ways and added new exception condtion cases to have a better exception management.